### PR TITLE
Refactor LOADING_ELEMENTS into a Set

### PR DESCRIPTION
**Pull request by jridgewell**
Partial for #36754

### CODE REVIEW

**Review by samouri — APPROVED** [review]

File: src/core/dom/layout/index.js [hunk 1]
```diff
@@ -51,26 +52,26 @@ export let DimensionsDef;
...
-  'AMP-IFRAME': true,
-  'AMP-IMG': true,
-  'AMP-INSTAGRAM': true,
-  'AMP-LIST': true,
-  'AMP-PINTEREST': true,
-  'AMP-PLAYBUZZ': true,
-  'AMP-RENDER': true,
-  'AMP-TIKTOK': true,
-  'AMP-TWITTER': true,
+export const LOADING_ELEMENTS_ENUM = {
```
[the remark this answers is not in the document]
Comment (jridgewell) [reply src/core/dom/layout/index.js:58]: Could be. I'll update.

**Comment by samouri** [comment]
LGTM

File: test/unit/test-custom-element.js [hunk 2]
```diff
@@ -2151,7 +2151,7 @@ describes.realWin('CustomElement', {amp: true}, (env) => {
         ElementClass = createAmpElementForTesting(win, TestElement);
         win.customElements.define('amp-test-loader', ElementClass);
         win.__AMP_EXTENDED_ELEMENTS['amp-test-loader'] = TestElement;
-        LOADING_ELEMENTS_ENABLED['amp-test-loader'.toUpperCase()] = true;
+        LOADING_ELEMENTS.add('amp-test-loader');
```
Comment (samouri) [inline test/unit/test-custom-element.js:2154]: Some unit tests are failing, is it possible we still need the `.toUpperCase()`?

**Comment by jridgewell** [comment]
It seem our minifier can't eliminate `Set([...])` calls, so I reverted back to a Enum.

**Pull request merged**